ethereum/contracts/ERC20App.sol

```diff
@@ -78,23 +79,30 @@ contract ERC20App is AccessControl {
         bytes32 _recipient,
         uint256 _amount,
         ChannelId _channelId,
-        uint32 _paraId
+        uint32 _paraId,
+        uint256 _weight
```

Comment:
Two things - 

1. Naming. We're actually asking the user to specifying the maximum DOT fee the recipient should pay to buy XCM execution weight. Fees and xcm weight are in different units so its important to make that distinction. So I would rename this to `_fee`/`fee` here and everywhere else.
2. Should use 16-byte`uint128`, not `uint256`, since we are specifying a quantity that only exists on the remote parachain, where DOT balances are recorded using `u128`. Using uint256 complicates things.